OnProfileUpdate/handler.ts

```diff
@@ -0,0 +1,188 @@
+import { pipe, flow } from "fp-ts/lib/function";
+import * as O from "fp-ts/lib/Option";
+import * as A from "fp-ts/ReadonlyArray";
+import * as E from "fp-ts/lib/Either";
+import * as T from "fp-ts/lib/Task";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as RTE from "fp-ts/lib/ReaderTaskEither";
+import { NonNegativeInteger } from "@pagopa/ts-commons/lib/numbers";
+import {
+  IProfileEmailReader,
+  IProfileEmailWriter,
+  ProfileEmail
+} from "@pagopa/io-functions-commons/dist/src/utils/unique_email_enforcement";
+import {
+  ProfileModel,
+  Profile
+} from "@pagopa/io-functions-commons/dist/src/models/profile";
+import { generateVersionedModelId } from "@pagopa/io-functions-commons/dist/src/utils/cosmosdb_model_versioned";
+import { CosmosErrors } from "@pagopa/io-functions-commons/dist/src/utils/cosmosdb_model";
+import { Logger } from "@azure/functions";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { PathReporter } from "io-ts/PathReporter";
+import { FiscalCode } from "../generated/backend/FiscalCode";
+
+const ProfileDocument = t.intersection([
+  ProfileEmail,
+  t.type({
+    _self: NonEmptyString,
+    isEmailValidated: t.boolean,
+    version: NonNegativeInteger
+  })
+]);
+
+type ProfileDocument = t.TypeOf<typeof ProfileDocument>;
+
+interface IDependencies {
+  readonly dataTableProfileEmailsRepository: IProfileEmailReader &
+    IProfileEmailWriter;
+  readonly profileModel: ProfileModel;
+  readonly logger: { readonly error: Logger["error"] };
+}
+
+const getPreviousProfile = (
+  fiscalCode: FiscalCode,
+  version: NonNegativeInteger
+) => ({
+  profileModel
+}: IDependencies): TE.TaskEither<CosmosErrors, O.Option<Profile>> =>
+  pipe(
+    version - 1,
+    NonNegativeInteger.decode,
+    E.fold(
+      () => TE.right(O.none),
+      previousVersion =>
+        pipe(
+          generateVersionedModelId<Profile, "fiscalCode">(
+            fiscalCode,
+            previousVersion
+          ),
+          id => profileModel.find([id, fiscalCode])
+        )
+    )
+  );
+
+const deleteProfileEmail = (profileEmail: ProfileEmail) => ({
+  dataTableProfileEmailsRepository
+}: IDependencies): TE.TaskEither<Error, void> =>
+  TE.tryCatch(
+    () => dataTableProfileEmailsRepository.delete(profileEmail),
+    error =>
+      error instanceof Error
+        ? error
+        : new Error("error deleting ProfileEmail from table storage")
+  );
+
+const insertProfileEmail = (profileEmail: ProfileEmail) => ({
+  dataTableProfileEmailsRepository
+}: IDependencies): TE.TaskEither<Error, void> =>
+  pipe(
+    TE.tryCatch(
+      () => dataTableProfileEmailsRepository.insert(profileEmail),
+      error => error
+    ),
+    TE.orElse(insertError =>
+      TE.tryCatch(
+        () =>
+          // check if the insert operation failed because the record was already there (for example in case of retry of the entire batch)
+          dataTableProfileEmailsRepository.get(profileEmail).then(() => {}),
```

Comment:
What if we use instead this instead an empty arrow function?
```suggestion
          dataTableProfileEmailsRepository.get(profileEmail).then(() => void 0),
```

Reply:
Fixed in 2c71684b0c5a50c169a19da7aee4ab4c2ff6870e.
Sorry :D